### ISSUE TITLE
Wrong Microsoft Notification Count (oauth errors)

### ISSUE THREAD

**Issue by arosboro** [wavebox/waveboxapp#358]
* Which version of Wavebox are you using? [e.g. 3.0.0]
Wavebox version 3.2.3 (beta channel)

* Which Operating System are you using? [e.g. Windows 10, Mac OSX 10, Ubuntu 14.04]
macOS Sierra Version 10.12.6

I see 4 unread notifications for office 365, but when I click the icon and open the unread email Wavebox opens a window that the item could not be found on Microsoft's application.

I have reloaded, resynced.  Changed the notification indicator setting, and re-authenticated.  I also tried turning off persist cookies.  I've searched my inbox for isread:false and there were no results.

When I open Developer Tools and the console, there are 401 (Unauthorized) errors trying to post to https://login.microsoftonline.com/common/oauth2/v2.0/token from mailboxes.js:66

```
Response
body: ReadableStream
bodyUsed: false
headers: Headers
__proto__: Headers
ok: false
redirected: false
status: 401
statusText: "Unauthorized"
type: "cors"
url: "https://login.microsoftonline.com/common/oauth2/v2.0/token"
__proto__: Response
```

**Comment by Thomas101** [comment]
Hi, thanks for reporting. When you say clicked the icon is this from the unread message in the macOS menu bar? Would you be able to send a screenshot of the error page that is displayed?

If you try adding your account a second time does that work or do you see the same behaviour? Besides the incorrect notification are you able to login and use your microsoft account okay?

**Comment by arosboro** [comment]
I have tried removing and adding the account, and am currently logged in with nothing out of the ordinary except being unable to un-flag these 4 messages.  (It still works).

![screen shot 2017-08-17 at 8 35 10 am](https://user-images.githubusercontent.com/2224595/29412524-3035368a-8327-11e7-889d-56f50724ff0f.png)

**Comment by Thomas101** [comment]
Thanks for trying that! It's weird because the url that Wavebox takes you to is the one that office sends to us. We're not doing anything magical with it :-/

There's a couple of things to try, to see if you can find the messages in case they've gotten lost on Office's end.

1. When you open the menu bar you should see the sender and subject of the email, for example `Thomas Beverley: test`...
![screen shot 2017-08-17 at 14 57 33](https://user-images.githubusercontent.com/103586/29415926-81af69ec-835c-11e7-9e34-3d69138e447c.png)
...if you try searching your inbox for `subject:my subject` do any of the emails come up?

2. When you come up against this window, if you try signing in to Office using your browser and then clicking the `Open in Browser` button (top right), does the email load or do you get the same message in the browser?

**Comment by arosboro** [comment]
I tried opening the link in my browser, which had the same alert box while logged into microsoftonline.

I've search for 2/4 emails using `from:` and `subject:` to find the email thread which was read.  I archived all of the messages in the 2 threads to see what would happen, and nothing changed in the indicator after reload/resyncing the account.

**Comment by arosboro** [comment]
I deleted and re/added my account this morning and the indicator works now.  I realized this is an office365 account and not an microsoft online account.  Initially the reason for deleting the account was that I was getting an error that I could not read emails at all.

**Comment by Thomas101** [comment]
Awesome! I think there is a lingering bug with this. We've just pushed a fix out in 3.2.5 beta for it. If it's all working for you again, am I okay to close this off?

**Comment by arosboro** [comment]
Yes you may close this.  I may have received the Beta update as that is the software channel I am using.
